procrastinate/contrib/django/django_connector.py

```diff
@@ -0,0 +1,207 @@
+        *,
+        json_dumps: Callable = None,
+        json_loads: Callable = None,
+        **kwargs: Any,
+    ):
+        self.json_dumps = json_dumps
+        self.json_loads = json_loads
+
+    def open(self, _=None) -> None:
+        pass
+
+    def close(self) -> None:
+        pass
+
+    async def open_async(self, _=None) -> None:
+        pass
+
+    async def close_async(self) -> None:
+        pass
+
+    def _wrap_json(self, arguments: dict[str, Any]):
+        return {
+            key: Json(value, dumps=self.json_dumps)
+            if isinstance(value, dict)
+            else value
+            for key, value in arguments.items()
+        }
+
+    @contextlib.contextmanager
+    def _connection(self) -> Iterator[ConnectionProxy]:
+        yield django_connection
+
+    def _dict_fetchone(self, cursor) -> dict[str, Any]:
+        """Return row from a cursor as a dict"""
+        columns = [col[0] for col in cursor.description]
+        r = dict(zip(columns, cursor.fetchone()))
+        try:  # TODO fix workaround
+            r["args"] = json.loads(r["args"])
+        except (KeyError, JSONDecodeError, TypeError):
+            pass
+        return r
+
+    def _dict_fetchall(self, cursor) -> list[dict[str, Any]]:
+        """Return all rows from a cursor as a dict"""
+        columns = [col[0] for col in cursor.description]
+        r = [dict(zip(columns, row)) for row in cursor.fetchall()]
+        for el in r:  # TODO fix workaround
+            try:
+                el["args"] = json.loads(el["args"])
+            except (KeyError, JSONDecodeError, TypeError):
+                pass
+        return r
+
+    @wrap_exceptions
+    @sync_to_async
+    def execute_query_async(self, query: str, **arguments: Any) -> None:
+        with self._connection() as connection:
+            with connection.cursor() as cursor:
+                cursor.execute(
+                    PERCENT_PATTERN.sub("%%", query), self._wrap_json(arguments)
+                )
+
+    @wrap_exceptions
+    @sync_to_async
+    def _execute_query_connection(
+        self, query: str, connection: DatabaseWrapper, **arguments: Any
+    ) -> None:
+        with connection.cursor() as cursor:
+            cursor.execute("LISTEN procrastinate_any_queue;")  # TODO
+            # cursor.execute(
+            #     PERCENT_PATTERN.sub("%%", query), self._wrap_json(arguments)
+            # )
+
+    @wrap_exceptions
+    @sync_to_async
+    def execute_query_one_async(self, query: str, **arguments: Any) -> dict[str, Any]:
+        return self.execute_query_one(query, **arguments)
+
+    @wrap_exceptions
+    def execute_query_one(self, query: str, **arguments: Any) -> dict[str, Any]:
+        with self._connection() as connection:
+            with connection.cursor() as cursor:
+                cursor.execute(
+                    PERCENT_PATTERN.sub("%%", query), self._wrap_json(arguments)
+                )
+                return self._dict_fetchone(cursor)
```

Comment:
Django cursor returns _str_ instead of _dict_ for _args_ field (jsonb). What is the cleanest solution for this?